src/utils/utils.js

```diff
@@ -67,15 +66,44 @@ export const getConfigObjectsForAnalytics = (configurations, groupCode) => {
     })
 
     // Create an object to index datasets by their IDs
-    const indexedDatasets = {};
+    const indexedDatasets = {}
 
     datasets.forEach((dataset) => {
-        indexedDatasets[dataset.id] = dataset;
-    });
+        indexedDatasets[dataset.id] = dataset
+    })
+
+    // filter out numerator relations where A or B is not in
+    // then map A, B numerators to ids
+    // then filter out any numerator relations where an ID is missing
+    const numeratorsInGroupCodes = numeratorsInGroup.map((num) => num.code)
+    const numeratorRelations = configurations.numeratorRelations
+        .filter(
+            (nr) =>
+                numeratorsInGroupCodes.includes(nr.A) ||
+                numeratorsInGroupCodes.includes(nr.B)
+        )
```

Comment:
Based on the "filter out numerator relations where A or B is not in", should the boolean operator be `&&` here? (i.e. if we want to filter _out_ NRs with either A or B missing, we want to filter _in_ the opposite, which would be NRs with both A and B present)

Reply:
Thanks. I think I forgot to finish writing out that comment.

The logic is correct (include numerator relations where either A or B is in the associated group). I've updated the comment to be clearer: `// filter numerator relations to include those for which A or B is in group`

Reply:
I think this is good. I can be merged and just resolve one conflict in utils.js